### PR TITLE
fix memory leak in Rank

### PR DESCRIPTION
**Pull request by homezcx**

### CODE REVIEW

File: rlclientlib/vw_model/safe_vw.cc [hunk 1]
```diff
@@ -142,6 +142,10 @@ namespace reinforcement_learning {
     for (auto&& ex : examples)
       _example_pool.emplace_back(ex);
 
+    for (auto e : examples) {
+      e->pred.a_s.delete_v();
+    }
```
Comment (rajan-chari) [inline rlclientlib/vw_model/safe_vw.cc]: Why not combine this into one loop?

Comment (homezcx) [reply rlclientlib/vw_model/safe_vw.cc]: :( didn't notice that. Will update.

File: rlclientlib/vw_model/safe_vw.cc [hunk 2]
```diff
@@ -138,9 +138,11 @@ namespace reinforcement_learning {
       scores[i] = predictions[i].score;
     }
 
-    // push examples back into pool for re-use
-    for (auto&& ex : examples)
+    // push examples back into pool for re-use and clean up examples
+    for (auto&& ex : examples) {
       _example_pool.emplace_back(ex);
+      ex->pred.a_s.delete_v();
```
Comment (rajan-chari) [inline rlclientlib/vw_model/safe_vw.cc]: Clean up the example before putting it back in the pool?

Comment (homezcx) [reply rlclientlib/vw_model/safe_vw.cc]: code and comment are updated.

**Comment by homezcx** [comment]
@rajan-chari  @ataymano @jackgerrits ping to see if we are good to merge.

**Pull request merged**